merino/providers/weather/backends/accuweather.py

```diff
@@ -92,6 +110,87 @@ def cache_inputs_for_weather_report(self, geolocation: Location) -> Optional[byt
 
         return (geolocation.country + geolocation.postal_code).encode("utf-8")
 
+    def cache_key_for_accuweather_request(
+        self, url: str, query_params: dict[str, str] = {}
+    ) -> str:
+        """Get the cache key for the accuweather request.
+        Also ensure that the API key is stripped out.
+        """
+        params = query_params.copy()
+        if "apikey" in params:
+            del params["apikey"]
```

Comment:
```suggestion
        params.pop(settings.accuweather.url_param_api_key, None)
```

nitpick: If it makes sense, let's use `settings.accuweather.url_param_api_key` here, instead of hard-coding `"apikey"`?

Reply:
Just a note that accessing a deeply nested property of the Dynaconf's settings object can be slow, in particular on the hot code paths as it will do multiple lookups. We've been bitten by that before. As a workaround, we often use a module constant or a class member to store its value for future reference.

```
URL_PARAM_API_KEY: str = settings.accuweather.url_param_api_key
```